web/src/components/CreateTestPlugins/Rest/steps/RequestDetails/BodyField/useBodyMode.tsx

```diff
@@ -0,0 +1,38 @@
+import {Dispatch, SetStateAction, useEffect, useState} from 'react';
+import Validator from 'utils/Validator';
+
+export type BodyMode = 'json' | 'xml' | 'raw' | 'none';
+
+function useGuessBodyModeEffect(
+  isEditing: undefined | boolean,
+  setBodyMode: Dispatch<SetStateAction<BodyMode>>,
+  body?: string
+) {
+  const [initialized, setInitialized] = useState(false);
+  useEffect(() => {
+    if (!initialized) {
```

Comment:
nested ifs can be usually simplified by having an `and` operation with a single if like:

`if(!initialized && isEditing && body)`

Then maybe we can have the rest of the validation logic at the Validator utility level and we can call it like `getBodyType(body)`